external/kafka/src/main/scala/org/apache/spark/streaming/kafka/NewDirectKafkaInputDStream.scala

```diff
@@ -0,0 +1,134 @@
+/*
```

Comment:
Hi Mario, I am following what Hadoop did when it was doing something similar. See NewHadoopRDD, for example. I am actually pretty flexible about the name so if folks disagree and have a better name to suggest, I am totally down for that. I just don't want to rename over and over again, so I'd suggest keeping the name the way it is right now, and get a consensus on what, if any, the new name should be. Once we have it here, I can change it.